build.rs

```diff
@@ -0,0 +1,85 @@
+
+use maplit::hashmap;
+use memofs::VfsSnapshot;
+
+fn snapshot_from_fs_path(path: &PathBuf) -> Result<VfsSnapshot, PathBuf> {
+    if path.is_dir() {
+        let entries: Result<Vec<fs::DirEntry>, PathBuf> = fs::read_dir(path)
+            .map_err(|_| path.to_owned())?
+            .map(|entry| entry.map_err(|_| path.to_owned()))
+            .into_iter()
+            .collect();
+
+        let vfs_entries: Result<Vec<(String, VfsSnapshot)>, PathBuf> = entries?
+            .iter()
+            .filter(|entry| {
+                let file_name = entry.file_name();
+                !file_name.to_string_lossy().ends_with(".spec.lua")
+            })
+            .map(|entry| {
+                let path = entry.path();
+
+                entry
+                    .file_name()
+                    .to_str()
+                    .ok_or(path.to_owned())
+                    .and_then(|file_name| {
+                        snapshot_from_fs_path(&path)
+                            .map(|snapshot| (file_name.to_owned(), snapshot))
+                    })
+            })
+            .into_iter()
+            .collect();
+
+        Ok(VfsSnapshot::dir(vfs_entries?))
+    } else {
+        println!("cargo:rerun-if-changed={}", path.to_string_lossy());
+        fs::read_to_string(path)
+            .ok()
+            .map(|content| VfsSnapshot::file(content))
+            .ok_or(path.to_owned())
+    }
+}
+
+fn main() {
+    let out_dir = env::var_os("OUT_DIR").unwrap();
+
+    let root_dir = env::var_os("CARGO_MANIFEST_DIR").unwrap();
+    let plugin_root = PathBuf::from(root_dir).join("plugin");
+
+    let plugin_modules = plugin_root.join("modules");
+
+    let snapshot = VfsSnapshot::dir(hashmap! {
+        "default.project.json" => snapshot_from_fs_path(&plugin_root.join("default.project.json")).unwrap(),
+        "fmt" => snapshot_from_fs_path(&plugin_root.join("fmt")).unwrap(),
+        "http" => snapshot_from_fs_path(&plugin_root.join("http")).unwrap(),
+        "log" => snapshot_from_fs_path(&plugin_root.join("log")).unwrap(),
+        "src" => snapshot_from_fs_path(&plugin_root.join("src")).unwrap(),
+        "modules" => VfsSnapshot::dir(hashmap! {
+            "roact" => VfsSnapshot::dir(hashmap! {
+                "src" => snapshot_from_fs_path(&plugin_modules.join("roact").join("src")).unwrap()
+            }),
+            "promise" => VfsSnapshot::dir(hashmap! {
+                "lib" => snapshot_from_fs_path(&plugin_modules.join("promise").join("lib")).unwrap()
+            }),
+            "t" => VfsSnapshot::dir(hashmap! {
+                "lib" => snapshot_from_fs_path(&plugin_modules.join("t").join("lib")).unwrap()
+            }),
+            "rbx-dom" => VfsSnapshot::dir(hashmap! {
+                "rbx_dom_lua" => VfsSnapshot::dir(hashmap! {
+                    "src" => snapshot_from_fs_path(&plugin_modules.join("rbx-dom").join("rbx_dom_lua").join("src")).unwrap()
+                })
+            }),
+        }),
+    });
```

Comment:
We talked in Discord about this block, and I've been thinking about it for awhile. I think it's okay to keep around for now, and we can revisit this problem later.